services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/taydennys/TaydennysServiceITest.kt

```diff
@@ -1026,6 +1031,77 @@ class TaydennysServiceITest(
                 alluClient.getApplicationInformation(hakemus.alluid!!)
             }
         }
+
+        @Test
+        fun `sends a new haittojenhallintasuunnitelma to Allu when the hakemus is a kaivuilmoitus`() {
+            val hanke = hankeFactory.builder(USERNAME).withHankealue().save()
+            val hankeEntity = hankeRepository.getReferenceById(hanke.id)
+            val hakemus =
+                hakemusFactory
+                    .builder(USERNAME, hankeEntity, ApplicationType.EXCAVATION_NOTIFICATION)
+                    .withMandatoryFields(hankealue = hanke.alueet.single())
+                    .withStatus(ApplicationStatus.WAITING_INFORMATION, alluId)
+                    .saveEntity()
+            val taydennys = taydennysFactory.builder(hakemus).withStartTime(startTime).save()
+            justRun { alluClient.respondToInformationRequest(any(), any(), any(), any()) }
+            every { alluClient.getApplicationInformation(any()) } returns
+                AlluFactory.createAlluApplicationResponse(
+                    hakemus.alluid!!,
+                    status = ApplicationStatus.HANDLING,
+                )
+            val attachments = mutableListOf<Attachment>()
+            every { alluClient.addAttachment(hakemus.alluid!!, capture(attachments)) } throws
+                AlluException()
```

Comment:
Why does this test throw Exception in sending attachment to Allu?

Reply:
It looks like I changed the wrong test after copy-pasting. Fixed now. Thank you for noticing.